pkg/statistics/histogram.go

```diff
@@ -1243,9 +1243,78 @@ func (b *bucket4Merging) Clone() bucket4Merging {
 	return *result
 }
 
+// mergeBucket is a special case when we don't have the NDV info for buckets.
+// We don't need to deal with the NDV info.
+func mergeBucket(sc *stmtctx.StatementContext, left *bucket4Merging, right *bucket4Merging) (*bucket4Merging, error) {
```

Comment:
Another suggestion is that `mergeBucket` and `mergeBucketNDV` seem very similar, could we merge them into one, which makes this part of code easier to maintain in the future.